specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/mfe.json

```diff
@@ -0,0 +1,4932 @@
+      },
+      "additionalProperties": false
+    },
+    "RootError": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "type": "string"
+        },
+        "severity": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "message": {
+          "type": "string"
+        },
+        "messageFormat": {
+          "type": "string"
+        },
+        "messageParameters": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "referenceCode": {
+          "type": "string"
+        },
+        "detailsUri": {
+          "format": "uri",
+          "type": "string"
+        },
+        "target": {
+          "type": "string"
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RootError"
+          }
+        },
+        "innerError": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/InnerErrorResponse"
+            }
+          ]
+        }
+      },
+      "additionalProperties": false
+    },
+    "CodeContainerResourceArmPaginatedResult": {
+      "description": "A paginated list of CodeContainer entities.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "An array of objects of type CodeContainer.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CodeContainerResource"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "AssetPath": {
+      "description": "Details of an AssetUri.",
+      "required": [
+        "path"
+      ],
+      "type": "object",
+      "properties": {
+        "path": {
+          "description": "The path of file/directory.",
+          "pattern": "\\w",
+          "type": "string"
+        },
+        "isDirectory": {
+          "description": "Whether the path defines a directory or a single file.",
+          "type": "boolean"
+        }
+      },
+      "additionalProperties": false
+    },
+    "CodeVersion": {
+      "description": "Details of an RegisteredCode",
+      "type": "object",
+      "properties": {
+        "datastoreId": {
+          "description": "The asset datastoreId",
+          "type": "string"
+        },
+        "assetPath": {
+          "description": "Details of asset path in the datastore.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/AssetPath"
+            }
+          ]
+        },
+        "description": {
+          "description": "The asset description text.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Tag dictionary. Tags can be added, removed, and updated.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
```

Comment:
why do most definitions have this open ended property bag?

Reply:
it's meant as a field that is named "properties" that lives within the resource's object. ie
```c#
public class MyClass
{
  public Dictionary<string, string> Description { get; set; };
  public Dictionary<string, string> Properties { get; set; };
  public Dictionary<string, string> Tags { get; set; };
  public AssetPath AssetPath { get; set;};
  ...
}
```